### ISSUE TITLE
Unity Livepeer Client

### ISSUE THREAD

**Issue by alextitonis** [livepeer/Grant-Program#128]
**Give a 3 sentence description for your proposal.**
Unity Livepeer Client for and AR Example

**Describe the problem you are solving.** 
A library for Unity 3D users and an example implementation using Vuforia AR in unity - Marker Based tracking & show video from livepeer

**Describe the solution you are proposing and how it will have a positive impact on the Livepeer developer ecosystem.** 
Will create the library in unity implementing the Livepeer API and will create some examples for other users

**Describe why you are the right team with the capability to build this.**
I have much experience in Unity, have worked for years in Unity3D and 2D projects, also in VR/AR games.

**Describe the scope of the project including a 3 month timeline and milestones.** 
There will be 2 milestones, the first one will be to create the library for the connection and the second will be the usage examples.

**Please estimate hours spent on project based on the above and how much funding you will need.** 
Around 1 month and a half, working on it half-time
The funding needed for the project is around 2.000-2.500$

**Comment by hansy** [comment]
Hi @alextitonis! Thanks for applying for a grant. I'd love to know a little more about the Unity Livepeer Client you have in mind to develop. Would this be a Unity wrapper around Livepeer APIs endpoints?

**Comment by alextitonis** [comment]
Exactly and planning to use it afterwords for an AR app i'm developing combined with clothes

**Comment by hansy** [comment]
@alextitonis Gotcha. Ok cool, we're definitely interested in the Unity library for Livepeer. Are you targeting all of Livepeer Studio's REST endpoints? Given Livepeer Studio is a developer-focused product, our primary goal is clear documentation for developers. The complimentary AR example is wonderful, but the first point of interface for most developers will be the docs. If you can provide clear documentation on how Unity developers can get started and use Livepeer, we'd love to offer a microgrant of $2K to help support this initiative. We'll also help get feedback on your docs from other developers. Lastly, do you see yourself being able to maintain these docs for the future?

**Comment by alextitonis** [comment]
Yeah we can work on the implementation in Unity, with docs.
Yeah we aim to maintain them as much as possible, as we want to use Livepeer in our project as well in the future

**Comment by alextitonis** [comment]
Wallet Address: 0x5204Bf4F2f24270561be48Ba38f71d1385BC839f
Febraury 25 - March 8

**Comment by hansy** [comment]
@alextitonis Upfront payment just went out (https://arbiscan.io/tx/0xb5bc9dfe4d61998f6e65fb0445a32a9807c3eb76ca7bea2f5f2a76a3e1b0e5c5). The upfront payment is 20%, so upon completion of your milestone, we'll unlock the rest! We'll set your start and end dates to Feb. 25, 2023 to Mar. 25, 2023.

**Comment by alextitonis** [comment]
Hey, got a kinda dumb question, i'm in trust wallet and Arbitrum count is still 0, should i add a different token?

**Comment by hansy** [comment]
@alextitonis The amount paid is in LPT, not arb eth. Let me know if you can see it!

**Comment by alextitonis** [comment]
I see, haven't received yet though (I use trust wallet).
![328614463_914481923019005_440336364336703408_n](https://user-images.githubusercontent.com/45359358/221374514-e555544a-9713-452b-86c4-74d6d7a516ac.jpg)

**Comment by hansy** [comment]
Is the wallet network set to arbitrum?

**Comment by alextitonis** [comment]
yes

**Comment by hansy** [comment]
@alextitonis Mmm can you try with another wallet? I'm not super familiar with Trust Wallet.

**Comment by alextitonis** [comment]
Sure, which wallet would you suggest?

**Comment by alextitonis** [comment]
Added it in metamask 0x3df573185CaD7bea3AcA6D2AD513fdF8C0b660f9
![Screenshot_587](https://user-images.githubusercontent.com/45359358/222370393-c640d7c4-bddd-4421-8764-ddea1c21bc20.png)

Which settings, should i put there?
![Screenshot_588](https://user-images.githubusercontent.com/45359358/222371203-6da273e1-b7e7-4d88-8a52-b02880bd7080.png)

**Comment by hansy** [comment]
We sent the funds to this wallet address (0x5204Bf4F2f24270561be48Ba38f71d1385BC839f), so if you're using metamask, you'll need to import that wallet address's seed phrase in.

**Comment by alextitonis** [comment]
This address was the trust wallet, but haven't received anything, as mentioned before

**Comment by hansy** [comment]
You should be able to import your trust wallet into metamask.

**Comment by alextitonis** [comment]
Done, could oyu send me the token contract and token decimal please?

**Comment by hansy** [comment]
Awesome, great to hear! How does the end of the month sound for completion of the grant?

**Comment by alextitonis** [comment]
Would it be possible to add one more week? because i'll be away from 20-25 of Match?

**Comment by hansy** [comment]
Yeah no worries! Send me a date that works for you so I can have it for my records.

**Comment by alextitonis** [comment]
Saturday 8 of April, if that's ok

**Comment by alextitonis** [comment]
Hey, here is a result, please let me know if you need me to add/change something or if it's ok
https://github.com/alextitonis/Unity-Livepeer-Client

**Comment by hansy** [comment]
Hey @alextitonis. Thanks for the update! What Livepeer endpoints are currently supported? I think I only see 5 endpoints documented in the README?

**Comment by alextitonis** [comment]
These were the ones that i found when foudn the doc, might be wrong, which ones are missing? I can add them

**Comment by hansy** [comment]
Yeah we have a few: https://docs.livepeer.org/reference/api

**Comment by alextitonis** [comment]
Thanks, will update it

**Comment by alextitonis** [comment]
Finished with all Routes, https://github.com/alextitonis/Unity-Livepeer-Client didn't write for each route, they are exactly the same, wrote for different ones, that take full data classes as inputs, to help users

**Comment by hansy** [comment]
This is great! I looked through your code and saw some of the new endpoints you added. Can you document those endpoints in the README? I was a little confused because the README only shows a few example endpoints. For devs, documentation is the most important thing. I think once the initial documentation is in place for all the endpoints you added, we can close out this grant. Also, are you part of any unity groups? I think the repo is in a good spot to start getting feedback from other devs as well as open up contributions from others. Great work!

**Comment by alextitonis** [comment]
Let me know when you check it please

**Comment by hansy** [comment]
This looks awesome Alex! Let me share this with my team to get their feedback!

**Comment by alextitonis** [comment]
Great thank you

**Comment by hansy** [comment]
Hey! I shared your Github repo in our Discord (https://discord.com/channels/423160867534929930/1051963444598943784/1102669773835341885). Please feel free to tag yourself to solicit feedback!

I think we can close out your grant soon. Give my team a couple more days to check out your repo, and then I can sign off!

**Comment by hansy** [comment]
Hey Alex, I'm going to queue up your last milestone payment. I'll update you when the tx is confirmed!

**Comment by hansy** [comment]
Your payment has been confirmed! https://arbiscan.io/tx/0xd23069e2c17c1cc2755b9956b953b1ccc71a4c53f957d50cd3973d1551cd5002

Closing this grant out. Amazing work @alextitonis! Let me know if there's anything else we can help with or if you have any additional questions!

**Comment by alextitonis** [comment]
thank you
